CRM/Core/DAO.php

```diff
@@ -3117,25 +3157,47 @@ protected static function getDynamicFkAclClauses(): array {
    *
    * With acls from related entities + additional clauses from hook_civicrm_selectWhereClause
    *
-   * @param string $tableAlias
+   * DO NOT OVERRIDE THIS FUNCTION
+   *
+   * @TODO: ADD `final` keyword to function signature
+   *
+   * @param string|null $tableAlias
+   * @param string|null $entityName
+   * @param array $conditions
+   *   Values from WHERE or ON clause
    * @return array
    */
-  public static function getSelectWhereClause($tableAlias = NULL) {
+  public static function getSelectWhereClause($tableAlias = NULL, $entityName = NULL, $conditions = []) {
     $bao = new static();
-    if ($tableAlias === NULL) {
-      $tableAlias = $bao->tableName();
-    }
-    $clauses = [];
-    foreach ((array) $bao->addSelectWhereClause() as $field => $vals) {
-      $clauses[$field] = NULL;
-      if ($vals) {
-        $clauses[$field] = "(`$tableAlias`.`$field` " . implode(" AND `$tableAlias`.`$field` ", (array) $vals) . ')';
-        if (empty(static::getSupportedFields()[$field]['required'])) {
-          $clauses[$field] = "(`$tableAlias`.`$field` IS NULL OR {$clauses[$field]})";
+    $tableAlias = $tableAlias ?? $bao->tableName();
+    $entityName = $entityName ?? CRM_Core_DAO_AllCoreTables::getBriefName($bao::class);
```

Comment:
@colemanw I was just doing a local upgrade from 5.66@head to master@head. This failed:

```
$ cv upgrade:db -vv

Box Requirements Checker
========================

> Using PHP 7.4.29
> PHP is using the following php.ini file:
  WARNING: No configuration file (php.ini) used by PHP!

> Checking Box requirements:
  .....


 [OK] Your system is ready to run the application.



Fatal error: Cannot use ::class with dynamic class name in /Users/totten/bknix/build/dmaster/web/sites/all/modules/civicrm/CRM/Core/DAO.php on line 3173
```

Reply:
@totten is it a php version thing? Would it work better with `get_class()`?

Reply:
@colemanw Yeah, I think that's right. `get_class()` seemed to work locally. I encountered two spots with the issue. Sent a follow-up PR for those two. (27536)